OBSDOCS-953/TRACING-4062 Write documentation and technical enablement for journald component (#75257)
openshift/openshift-docs: modules/otel-collector-components.adoc

[id="otlp-receiver_{context}"]
=== OTLP Receiver

> The OTLP receiver ingests traces and metrics by using the OpenTelemetry Protocol (OTLP).

Reviewer: It can ingest logs as well

Wildcards with `+*+` are accepted under the `cors_allowed_origins`.
To match any origin, enter only `+*+`.

[id="journald-receiver_{context}"]
=== Journald receiver

> The Journald receiver is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

Reviewer 2: We just lower-case journald in the docs. 

```suggestion
The journald receiver is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```

Author: Thank you. I'm actually capitalizing all the component names now, so no need to worry about this one.

Author: However, there is one more instance that I am lowercasing. Thank you again.

    seLinuxOptions:
      type: spc_t
    seccompProfile:
      type: RuntimeDefault
  config: |
    receivers:
      journald:
        files: /var/log/journal/*/*
        priority: info # <1>
        units: # <2>
          - kubelet
          - crio
          - init.scope
          - dnsmasq
        all: true # <3>
        retry_on_failure:
          enabled: true # <4>
          initial_interval: 1s # <5>
          max_interval: 30s # <6>
          max_elapsed_time: 5m # <7>
    processors:
    exporters:
      debug:
        verbosity: detailed
    service:
      pipelines:
        logs:
          receivers: [journald]
          exporters: [debug]
  volumeMounts:
  - name: journal-logs
    mountPath: /var/log/journal/
    readOnly: true
  volumes:
  - name: journal-logs
    hostPath:
      path: /var/log/journal
  tolerations:
  - key: node-role.kubernetes.io/master
    operator: Exists
    effect: NoSchedule
EOF
----
<1> Filters output by message priorities or priority ranges. Defaults to `info`.
<2> Lists the units to read entries from. If empty, entries are read from all units.
<3> Includes very long logs and logs with unprintable characters. Defaults to `false`.

> <4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when encounters an error from downstream components. Defaults to `false`.

Reviewer 2: I think you are missing a word here. 
```suggestion
<4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when <MISSING WORD?>encounters an error from downstream components. Defaults to `false`.
```

Author: :slightly_smiling_face: thank you. I think it's meant to be `encountering`, not `encounters`. Fixed.

    allowPrivilegeEscalation: false
    capabilities:
      drop:
      - CHOWN
      - DAC_OVERRIDE
      - FOWNER
      - FSETID
      - KILL
      - NET_BIND_SERVICE
      - SETGID
      - SETPCAP
      - SETUID
    readOnlyRootFilesystem: true
    seLinuxOptions:
      type: spc_t
    seccompProfile:
      type: RuntimeDefault
  config: |
    receivers:
      journald:
        files: /var/log/journal/*/*
        priority: info # <1>
        units: # <2>
          - kubelet
          - crio
          - init.scope
          - dnsmasq
        all: true # <3>
        retry_on_failure:
          enabled: true # <4>
          initial_interval: 1s # <5>
          max_interval: 30s # <6>
          max_elapsed_time: 5m # <7>
    processors:
    exporters:
      debug:
        verbosity: detailed
    service:
      pipelines:
        logs:
          receivers: [journald]
          exporters: [debug]
  volumeMounts:
  - name: journal-logs
    mountPath: /var/log/journal/
    readOnly: true
  volumes:
  - name: journal-logs
    hostPath:
      path: /var/log/journal
  tolerations:
  - key: node-role.kubernetes.io/master
    operator: Exists
    effect: NoSchedule
EOF
----

> <1> Filters output by message priorities or priority ranges. Defaults to `info`.
> <2> Lists the units to read entries from. If empty, entries are read from all units.
> <3> Includes very long logs and logs with unprintable characters. Defaults to `false`.
> <4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when encountering an error from downstream components. Defaults to `false`.

Reviewer 3: ```suggestion
<1> The priorities or priority ranges to filters output by . The defaults is `info`.
<2> The list of units to read entries from. If empty, entries are read from all units.
<3> If the specified value is `true`, very long logs and logs with unprintable characters are included. The default is `false`.
<4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when encountering an error from downstream components. The default is `false`.
```

Author: Thank you. I applied some of these suggestions.

<5> The time interval to wait after the first failure before retrying. Defaults to `1s`. The units are `ms`, `s`, `m`, `h`.
<6> The upper bound for the retry backoff interval. When this value is reached, the time interval between consecutive retry attempts remains constant at this value. Defaults to `30s`. The supported units are `ms`, `s`, `m`, `h`.

> <7> The maximum time interval, including retry attempts, for attempting to send a logs batch to a downstream consumer. When this value is reached, the data are discarded. If the set value is `0`, retrying never stops. Defaults to `5m`. The supported units are `ms`, `s`, `m`, `h`.

Reviewer 3: ```suggestion
<7> The maximum time interval, including retry attempts, for attempting to send a logs batch to a downstream consumer. When this value is reached, the data are discarded. If the set value is `0`, retrying never stops. The defaults is `5m`. The supported units are `ms`, `s`, `m`, `h`.
```

Author: Thank you. Applied.
